codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerServiceGeneratorV2.kt

```diff
@@ -230,9 +238,32 @@ class ServerServiceGeneratorV2(
             "NullabilityChecks" to nullabilityChecks,
             "RoutesArrayElements" to routesArrayElements,
             "SmithyHttpServer" to smithyHttpServer,
+            "PatternInitializations" to patternInitializations(),
         )
     }
 
+    /**
+     * Renders `PatternString::compile_regex()` function calls for every
+     * `@pattern`-constrained string shape in the service closure.
+     */
+    @Suppress("DEPRECATION")
+    private fun patternInitializations(): Writable {
+        val patterns = Walker(model).walkShapes(service)
+            .filter { shape -> shape is StringShape && shape.hasTrait<PatternTrait>() && !shape.hasTrait<software.amazon.smithy.model.traits.EnumTrait>() }
+            .map { shape -> codegenContext.constrainedShapeSymbolProvider.toSymbol(shape) }
+            .map { symbol ->
+                writable {
+                    rustTemplate("#{Type}::compile_regex();", "Type" to symbol)
+                }
+            }.toMutableList()
+
+        if (patterns.isNotEmpty()) {
+            patterns += writable { rust("// Eagerly initialize regexes for `@pattern` strings.") }
```

Comment:
Extremely minor nit: it'd be nicer to prepend this comment, instead of appending it at the end.

Reply:
can do `patterns.letIf(patterns.isNotEmpty()) { docs + patterns }`

Reply:
Thank you both! This was an oversight, I intended to prepend it. Done in 8475b143a5cc680afb6a2ed6c883c6dc4cb23b90.